### PR TITLE
javascript/all-your-base: initial mentor notes

### PR DESCRIPTION
**Pull request by SleeplessByte**

### CODE REVIEW

**Comment by SleeplessByte** [comment]
Build failing has nothing to do with this PR.

**Comment by tejasbubane** [comment]
@SleeplessByte There is nothing checked in this build. I guess it was just bootstrapped with Ruby.

cc @iHiD

**Comment by SleeplessByte** [comment]
Yep, so you can just review. Jeremy will replace/fix Travis in a few days :)

File: tracks/javascript/exercises/all-your-base/mentoring.md [hunk 1]
```diff
@@ -0,0 +1,110 @@
+### Reasonable solution
+
+The most important thing is that the student has implement the algorithm(s)
```
Comment (tejasbubane) [inline tracks/javascript/exercises/all-your-base/mentoring.md]:
```suggestion
The most important thing is that the student has to implement the algorithm(s)
```

File: tracks/javascript/exercises/all-your-base/mentoring.md [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+export function convert(digits, base, nextBase) {
+  assertBase(base, 'input')
+  assertBase(nextBase, 'output')
+  assertDigits(digits, base)
+
+  return toBaseDigits(toBase10(digits, base), nextBase)
+}
+```
+
+There are various way to approach this exercise. In general, it should consist
```
Comment (tejasbubane) [inline tracks/javascript/exercises/all-your-base/mentoring.md]:
```suggestion
There are various ways to approach this exercise. In general, it should consist
```

File: tracks/javascript/exercises/all-your-base/mentoring.md [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+- convert the value to an arbitrary base
+
+Note: the two conversion steps _can_ be reduced to a single step, by
+calculating the "base" between the two. However, this is usually somewhat
+harder to comprehend. Take note that this is a possibility, and a valid one.
+
+#### Converting to base 10
+
+There are various approaches, but the two most concise ones, derived from
+compsci and mathematics, is as follows:
```
Comment (tejasbubane) [inline tracks/javascript/exercises/all-your-base/mentoring.md]:
```suggestion
compsci and mathematics, are as follows:
```

**Comment by tejasbubane** [comment]
I agree this is harder to comprehend common conversion approach. It makes sense users first do the `base10` function and then the `arbitrary` one. But I would always suggest removing the `base10` & reusing the more generic `arbitray` one passing `10` as the base argument. Should we add it to common suggestions?

**Comment by SleeplessByte** [comment]
Could you write up the sentences 🥇 ? I think it's fine to suggest indeed.

**Comment by iHiD** [comment]
So what's changed to make Travis start failing? It was being used here before IIRC.

**Comment by SleeplessByte** [comment]
Nothing. Check the log there, it's some gem failure. Seems Travis 
internal

**Comment by tejasbubane** [comment]
Found this https://travis-ci.community/t/bundle-install-started-failing-on-ruby-build/4368 but it was few months back. Not sure since when our builds started failing.

**Comment by iHiD** [comment]
Should be fixed. Can you rebase to test it pls?

File: tracks/javascript/exercises/all-your-base/mentoring.md [hunk 4]
```diff
@@ -0,0 +1,110 @@
+### Reasonable solution
```
Comment (SleeplessByte) [inline tracks/javascript/exercises/all-your-base/mentoring.md]:
```suggestion
### Reasonable solutions
```

**Comment by iHiD** [comment]
Thanks. Looks like we're back to normal.

**Pull request merged**